src/WebJobs.Script/Diagnostics/OpenTelemetry/OpenTelemetryConfigurationExtensions.cs

```diff
@@ -28,42 +28,34 @@ internal static class OpenTelemetryConfigurationExtensions
     {
         internal static void ConfigureOpenTelemetry(this ILoggingBuilder loggingBuilder, HostBuilderContext context, TelemetryMode telemetryMode)
         {
-            var connectionString = GetConfigurationValue(EnvironmentSettingNames.AppInsightsConnectionString, context.Configuration);
-            var (azMonConnectionString, credential, enableOtlp, enableAzureMonitor) = telemetryMode switch
-            {
-                // Initializing OTel services during placeholder mode as well to avoid the cost of JITting these objects during specialization.
-                // Azure Monitor Exporter requires a connection string to be initialized. Use placeholder connection string if in placeholder mode.
-                TelemetryMode.Placeholder => (
-                    "InstrumentationKey=00000000-0000-0000-0000-000000000000;",
-                    null,
-                    true,
-                    true),
-                _ => (
-                    connectionString,
-                    GetTokenCredential(context.Configuration),
-                    !string.IsNullOrEmpty(GetConfigurationValue(EnvironmentSettingNames.OtlpEndpoint, context.Configuration)),
-                    !string.IsNullOrEmpty(connectionString))
-            };
-
-            // If neither OTLP nor Azure Monitor is enabled, don't configure OpenTelemetry.
-            if (!enableOtlp && !enableAzureMonitor)
+            var otlpEndpoint = GetConfigurationValue(EnvironmentSettingNames.OtlpEndpoint, context.Configuration);
+            var azMonConnectionString = GetConfigurationValue(EnvironmentSettingNames.AppInsightsConnectionString, context.Configuration);
+
+            bool enableOtlp = !string.IsNullOrWhiteSpace(otlpEndpoint);
+            bool enableAzureMonitor = !string.IsNullOrWhiteSpace(azMonConnectionString);
+
+            TokenCredential credential = enableAzureMonitor ? GetTokenCredential(context.Configuration) : null;
```

Comment:
Can this be moved closer to usage? In the use case where we are exiting, there is no need to execute this line, I guess

Reply:
I will refactor this when I update the nuget package.